### PR TITLE
Use only stat, not access to determine readonly, when root.

### PR DESCRIPTION
**Pull request by jaykrell**
This is closer to how Windows works, closer to how CoreCLR appears,
and should fix one of GetAttributes_ReadOnly and Create_Path_ReadOnly running as root (LLVM CI).
It should fix GetAttributes_ReadOnly.

"Fixing" Create_Path_ReadOnly would would require File.Open/Create to synthesize the readonly bit
with stat (like here) and implements its own denial instead of just calling underlying
open/create to decide, thereby denying access when the underlying system would not.

See https://github.com/mono/mono/issues/8903.

### CODE REVIEW

**Comment by jaykrell** [comment]
See also https://github.com/mono/mono/pull/8904

File: mono/metadata/w32file-unix.c [hunk 1]
```diff
@@ -953,16 +946,9 @@ is_file_writable (struct stat *st, const gchar *path)
...
 		return 1;
 
-	/* Fallback to using access(2). It's not ideal as it might not take into consideration euid/egid
-	 * but it's the only sane option we have on unix.
-	 */
-	MONO_ENTER_GC_SAFE;
-	ret = access (path, W_OK) == 0;
-	MONO_EXIT_GC_SAFE;
-	return ret;
+	return 0;
```
Comment (lewurm) [inline mono/metadata/w32file-unix.c]: can we do the fallback when the runtime isn't executed as root?

**Comment by jaykrell** [comment]
Sure. It is all a bit nebulous. I’ve concluded the portable subset of file system features is tiny, and mono tries to exceed it.

**Comment by jaykrell** [comment]
Does access make a difference? Maybe handles extended functionality on Unix?

**Comment by jaykrell** [comment]
Better? This should bring “llvm” failures from 3 to 2.

**Comment by akoeplinger** [comment]
Went with https://github.com/mono/mono/pull/8904 instead.

**Pull request closed without merging**